### PR TITLE
Update Mixer Report() handling

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR accomplishes a number of cleanup items related to Mixer's handling of Report()s:
- adds example config for a `stdio` handler for `logentry` instances
- updates `stdio` log handling to support `IP_ADDRESS` information (as currently handled)
- updates the example config for metrics to resolve problematic naming and inclusion of `method`
- updates the `prometheus` and `stackdriver` metric configs to match
- updates the `grafana` dashboard config and `servicegraph` code to match new metrics shape
- adds a `filter_empty=true` option to `servicegraph` to allow display of only services that have received traffic in the `time_horizon`

In so doing, this PR addresses a number of outstanding reported issues, including:
- Review metrics dimensions (#768)
- switch from `target` to `destination`
- Recent user complaints about `servicegraph` including stale data
- Need for TCP service monitoring in `grafana`
- Need to support namespaced services everywhere
- `method` (as currently eval'd) leads to pollution of tsdb. 

New example screenshots:
- Dashboard (top):  https://storage.googleapis.com/istio-gists/dashboard-update-top.png
- Dashboard (http): https://storage.googleapis.com/istio-gists/dashboard-update-http.png
- Dashboard (tcp): https://storage.googleapis.com/istio-gists/dashboard-update-tcp.png

**Release note**:

```release-note
Update metrics, dashboard, and servicegraph to support destination and multiple namespaces.
```

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/1321)

### CODE REVIEW

File: adapter/stdio/stdio.go [hunk 1]
```diff
@@ -57,7 +58,18 @@ func (h *handler) HandleLogEntry(_ context.Context, instances []*logentry.Instan
 
 		for _, varName := range h.logEntryVars[instance.Name] {
 			if value, ok := instance.Variables[varName]; ok {
-				fields = append(fields, zap.Any(varName, value))
+				// TODO: remove when IP_ADDRESS is properly handled by Mixer
+				switch value.(type) {
```
Comment (mandarjog) [inline adapter/stdio/stdio.go]: consider converting this to
`func field(varName string, value interface{}) zapcore.Field { }`
and reusing it in `HandleLogEntry` and `HandleMetrics`

File: testdata/config/metrics.yaml [hunk 2]
```diff
@@ -6,11 +6,10 @@ metadata:
 spec:
   value: "1"
   dimensions:
-    source: source.labels["app"] | "unknown"
-    target: target.service | "unknown"
-    service: target.labels["app"] | "unknown"
-    method: request.path | "unknown"
-    version: target.labels["version"] | "unknown"
+    source_service: source.service | "unknown"
+    source_version: source.labels["version"] | source.labels["istio"] | "unknown"
```
Comment (ZackButcher) [inline testdata/config/metrics.yaml]: Why do we use the "istio" label for version?

Comment (douglas-reid) [reply testdata/config/metrics.yaml]: this a good catch. this slipped in when I was bouncing ideas off of people -- the dangers of not checking your current branch.

Comment (douglas-reid) [reply testdata/config/metrics.yaml]: Fixed.

File: adapter/stdio/stdio.go (hunk 1 above)
Comment (douglas-reid) [reply adapter/stdio/stdio.go]: done.

**Comment by douglas-reid** [comment]
PTAL.

**Pull request merged**